File: lectures/bonus/language_levels/std_dev.c
#include <stdio.h>

void main()
{
    int n = 6;
    int data[] = { 11, 12, 13, 14, 15, 16 };

    int total = 0;
    for (int i = 0; i < n; i++) {
        total += data[i];
    }
 
    float mean = total / n;
 
    float total_deviation = 0;
 
    for (int i = 0; i < n; i++) {
        total_deviation += (data[i] - mean) * (data[i] - mean);
    }
 
    float variance = total_deviation / n;
 
    float standardDeviation = variance / variance;
}

File: lectures/bonus/language_levels/std_dev.py

data = [11, 12, 13, 14, 15, 16]
 
mean = sum(data) / len(data)

total_deviation = 0
for elem in data:
    total_deviation += (elem - mean) ** 2

variance = total_deviation / len(data)

standardDeviation = variance / variance
